Add matplotlib plotting to VertexClustering

iosonofabio opened the pull request:
This PR aims to fix #370 

ATM there is no specific test for this.

ntamas commented:
It's hard to test anyway, so let's just roll with the implementation. Let me know when it's ready for review.

iosonofabio commented:
The whole "mark_groups" option is a little weird when used in conjunction with a `VertexClustering` as the first argument of `igraph.plot`, but changing that breaks API for the Cairo backend, so let's not touch it here.

Ready for merge.

vtraag commented:
> The whole "mark_groups" option is a little weird when used in conjunction with a `VertexClustering` as the first argument of `igraph.plot`, but changing that breaks API for the Cairo backend, so let's not touch it here.

Well, yes, sort of. A simple boolean would suffice, simply to indicate whether you want to draw the convex hulls or not.

iosonofabio commented:
Fair enough, I didn't know that was an option since it's not mentioned in the docs. I added it and also cleaned up the code a little, and added this info to the docs.

I then rebased on top of the current master, so if it passes CI it should be good to go.

ntamas commented on src/igraph/drawing/graph.py:
```diff
@@ -1010,6 +1012,61 @@ def callback_edge_offset(event):
+                del kwds["mark_groups"]
+            elif (kwds["mark_groups"] is True) and (clustering is not None):
+                pass
+            # Deferred import to avoid a cycle in the import graph
```
The deferred import is not here but above

ntamas commented on src/igraph/drawing/graph.py:
```diff
@@ -1010,6 +1012,61 @@ def callback_edge_offset(event):
+                kwds["vertex_color"] = c
+
+                # mark_groups if not explicitely marked
+                if ("mark_groups" in kwds) and (kwds["mark_groups"] is True):
```
`if kwds.get("mark_groups") is True` is enough; `.get()` returns `None` if the key is not in the dict

ntamas reviewed (approved):
Okay, seems good, only a few minor changes were requested but I'll do these myself.
